### PR TITLE
Upgrade to Gulp 4

### PR DESCRIPTION
**Pull request by marcokrikke**
Ref. eveoh/tasks#91
Ref. eveoh/tasks#95

## Proposed changes

Quite a big-bang upgrade, but everything is kind of related to each other.

- Upgraded to Gulp 4
- Fixed .npmrc so it uses the correct proxy on our end
- Added .bowerrc so it uses our proxy
- Added a package.json lockfile
- Upgrade to NodeJS latest LTS (required for Gulp 4 anyway)
- Fixed appcache.manifest generation
- Removed use of deprecated gulp-util package
- Replaced gulp-ruby-sass by gulp-sass and node-sass
- Removed use of gulp-sourcemaps and use built in support
- Generate sourcemaps for JS as well
- Fixed default config
- Upgraded Gradle to 5.6.4

## Testing

- Tested build and compared output to 'old' output (not minified). Exact match.
- Tested war, zip and zipVersioned artifacts. Exact match to old artifacts.
- Tested server, runs as expected.
- Tested watch, runs as expected after upgrade to node-sass
- Tested sourcemap generation

## Documentation

None.

## Upgrade notes for breaking changes

- [x] `UPGRADE_NOTES.md` has been updated (when applicable).

### CODE REVIEW

**Comment by marcokrikke** [comment]
@troley ik heb `bower-nexus3-resolver` overgeslagen, zie eveoh/mytimetable#4542

File: build.gradle [hunk 1]
```diff
@@ -83,10 +83,10 @@ uploadArchives.dependsOn gulp_zip
 
 node {
     // Version of node to use.
-    version = '6.9.4'
+    version = '12.13.1'
 
     // Version of npm to use.
-    npmVersion = '3.10.10'
+    npmVersion = '6.12.1'
```
Comment (troley) [inline build.gradle]: Latest versie van npm is 6.13.2. Is er een reden voor om liever 6.12.1 te gebruiken?

Comment (marcokrikke) [reply build.gradle]: Op https://nodejs.org/en/download/ staat:

> Latest LTS Version: 12.13.1 (includes npm 6.12.1)

Heb het dus maar op die twee versies gehouden

Comment (troley) [reply build.gradle]: Aha makes sense. Kan in dat geval beter 6.12.1 blijven inderdaad.

**Comment by troley** [comment]
Ziet er verder prima uit @marcokrikke . Mag ik hem mergen?

**Comment by marcokrikke** [comment]
> Ziet er verder prima uit @marcokrikke . Mag ik hem mergen?

Ja hoor 👍

**Pull request merged**